pytorch_lightning/utilities/enums.py

```diff
@@ -49,6 +49,29 @@ class AMPType(LightningEnum):
     NATIVE = "native"
 
 
+class PrecisionType(LightningEnum):
+    """Type of precision used.
+
+    >>> PrecisionType.HALF == 16
+    True
+    >>> PrecisionType.HALF in (16, "16")
+    True
+    """
+
+    HALF = 16
+    FLOAT = 32
+    FULL = 64
```

Comment:
Are you sure this works well?

`LightningEnum` subclasses `str, Enum` so the int values here might break things.

Convert them to str?

Reply:
Seemed to work fine (the doc test explicitly tests that an int comparison works), have converted them to string though for explicitness!